### DISCUSSION TITLE
Authenticate remote registry using GCloud private registry

### DISCUSSION THREAD

**Discussion by uncvrd** [caprover/caprover#1106] [Q&A]
Hi! I'm in the process of setting up CI/CD by using the Caprover docs, and instead of GitLab, I am using GCloud's cloudbuild to deploy private images to their google registry url `gcr.io`. In my captain dashboard, it shows that GCloud can be used as a remote registry. However, I am a bit confused on how to authenticate. Google provides us with the following authentication methods:

![image](https://user-images.githubusercontent.com/25097099/118388570-26008b80-b5da-11eb-9c02-64375881f578.png)

Reference: https://cloud.google.com/artifact-registry/docs/docker/authentication#standalone-helper

Obviously it would be great to use the access token method like GitLab, however gclouds access tokens are short-lived (60 minutes) so I'm a bit confused on how I set up a remote registry, especially when I do have to scale up using clusters, knowing when to re-auth before a new image needs to be pulled could cause an issue. 

What would be the best approach to authenticate with gcloud so that I can pull the remote image using the `--imageName` flag while running `caprover deploy`?

**Comment by githubsaturn** [comment]
Not sure how gcloud image registry works. But if it helps, all CapRover does is to pass username and pass in the header when it makes an API call to docker engine. 

Think of this as simple values that your provide to `docker login` command

**Reply by uncvrd** [reply]
Thanks @githubsaturn my current workaround is to deploy a docker `registry` image as an app on caprover, so that I'm not using the internal self-hosted registry. I've then connected a gcloud storage driver to my registry so I can store my images on Google Cloud. And from what you've mentioned, it sounds like I can utilize the `HTTP Basic Auth` you've integrated for apps on caprover to secure the registry and then just pass those credentials in to the remote registry config options?
